### PR TITLE
docs(split-button): refactor split-button stories to use csf and mdx - FE-3447

### PR DESCRIPTION
**Pull request by mkrds**
This PR refactors SplitButton stories to use `CSF` / `MDX` formats instead of deprecated `storiesOf`

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/split-button/split-button-children.style.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const StyledSplitButtonChildrenContainer = styled.div`
     min-width: 75%;
     white-space: nowrap;
     z-index: ${theme.zIndex.popover};
```
Comment (DipperTheDan) [inline src/components/split-button/split-button-children.style.js:12]: Two z-indexes here. Is this correct?

File: src/components/split-button/split-button-children.style.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const StyledSplitButtonChildrenContainer = styled.div`
...
+    z-index: 10000;
```
Comment (DipperTheDan) [inline src/components/split-button/split-button-children.style.js:13]: Two z-indexes here. Is this correct?

File: cypress/features/regression/splitButton.feature [hunk 3]
```diff
@@ -1,47 +1,9 @@
...
-
-  @positive
-  Scenario: I enable Split Button component
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "disabledFalse" object name
-    Then Button is enabled
-
-  @positive
-  Scenario Outline: I set Split Button text align to <align>
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "<nameOfObject>" object name
-    Then Split Button component additional buttons text align is set to "<align>" align
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:36]: remove this step from `buton-steps.js`

File: cypress/features/regression/splitButton.feature [hunk 4]
```diff
@@ -1,47 +1,9 @@
...
-
-  @positive
-  Scenario: I disable Split Button component
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "disabled" object name
-    Then Button is disabled
-
-  @positive
-  Scenario: I enable Split Button component
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "disabledFalse" object name
-    Then Button is enabled
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:31]: remove this step from `buton-steps.js`

File: cypress/features/regression/splitButton.feature [hunk 5]
```diff
@@ -1,47 +1,9 @@
...
-    Examples:
-      | size   | px | nameOfObject |
-      | small  | 32 | sizeSmall    |
-      | medium | 40 | sizeMedium   |
-      | large  | 48 | sizeLarge    |
-
-  @positive
-  Scenario: I disable Split Button component
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "disabled" object name
-    Then Button is disabled
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:26]: remove this step from `buton-steps.js`

File: cypress/features/regression/splitButton.feature [hunk 6]
```diff
@@ -1,47 +1,9 @@
...
-    Then Button background color is "<color>"
-    Examples:
-      | buttonType | color            | nameOfObject        |
-      | primary    | rgb(0, 129, 93)  | buttonTypePrimary   |
-      | secondary  | rgba(0, 0, 0, 0) | buttonTypeSecondary |
-
-  @positive
-  Scenario Outline: I select size to <size>
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "<nameOfObject>" object name
-    Then Split Button is set to "<size>" size and has <px> px height
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:16]: remove this step from `buton-steps.js`

File: cypress/features/regression/splitButton.feature [hunk 7]
```diff
@@ -1,47 +1,9 @@
 Feature: Split Button component
   I want to test Split Button component properties
 
-  @positive
-  Scenario Outline: I select buttonType to <buttonType>
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "<nameOfObject>" object name
-    Then Button background color is "<color>"
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:7]: remove this step from `buton-steps.js`

File: cypress/features/regression/splitButton.feature [hunk 8]
```diff
@@ -50,31 +12,22 @@ Feature: Split Button component
...
     Then Button subtext on preview is <subtext>
     Examples:
       | subtext                      | nameOfObject            |
       | mp150ú¿¡üßä                  | subtextOtherLanguage    |
       | !@#$%^*()_+-=~[];:.,?{}&"'<> | subtextSpecialCharacter |
 
-  @positive
-  Scenario Outline: I check icon positioning to <iconPosition>
-    When I open default "Split Button" component in noIFrame with "splitButton" json from "commonComponents" using "<nameOfObject>" object name
-    Then Split Button iconPosition is set to "<iconPosition>" and has "warning" icon
```
Comment (KatarzynaQA) [inline cypress/features/regression/splitButton.feature:63]: remove this step from `buton-steps.js`

**Review by KatarzynaQA — DISMISSED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Pull request merged**